tests/performance/run.mjs

```diff
@@ -84,110 +82,393 @@ let currentBrowser;
-  console.error("Error:", err);
-  return process.exit(1);
-});
-
-/** Initialize and start all tests on Chrome. */
-async function start() {
-  await initScripts();
-  await initServers();
-
-  onFinished = () => {
-    const hasSucceededOnChrome = compareSamples();
-    shutdown().catch((err) => {
-      // eslint-disable-next-line no-console
-      console.error("Failed to shutdown:", err);
-    });
-    if (!hasSucceededOnChrome) {
-      // eslint-disable-next-line no-console
-      console.error("Tests failed on Chrome");
-      return process.exit(1);
-    }
-    return process.exit(0);
-
-    // TODO also run on Firefox? Despite my efforts, I did not succeed to run
-    // tests on it.
-    // onFinished = async () => {
-    //   shutdown();
-    //   const hasSucceededOnFirefox = compareSamples();
-    //   if (!hasSucceededOnChrome || !hasSucceededOnFirefox) {
-    //     // eslint-disable-next-line no-console
-    //     console.error("Tests failed on:" +
-    //                   (!hasSucceededOnChrome ? " Chrome" : "") +
-    //                   (!hasSucceededOnFirefox ? " Firefox" : ""));
-    //     return process.exit(1);
-    //   }
-    //   return process.exit(0);
-    // };
-    // startAllTestsOnFirefox();
-  };
+    const closeServers = () => {
+      contentServer?.close();
+      contentServer = undefined;
+      resultServer?.close();
+      resultServer = undefined;
+      staticServer?.close();
+      staticServer = undefined;
+    };
 
-  startAllTestsOnChrome().catch((err) => {
-    // eslint-disable-next-line no-console
-    console.error("Error:", err);
-    return process.exit(1);
+    initServers(onFinished, onError)
+      .then((servers) => {
+        contentServer = servers.contentServer;
+        resultServer = servers.resultServer;
+        staticServer = servers.staticServer;
+        if (isFinished) {
+          closeServers();
+        }
+        return startAllTestsOnChrome();
+      })
+      .catch(onError);
   });
 }
 
 /**
  * Initialize all servers used for the performance tests.
+ * @param {Function} onFinished
+ * @param {function} onError
  * @returns {Promise} - Resolves when all servers are listening.
  */
-async function initServers() {
-  const contentServer = createContentServer(CONTENT_SERVER_PORT);
-  const staticServer = launchStaticServer(currentDirectory, {
-    httpPort: PERF_TESTS_PORT,
-  });
-  const resultServer = createResultServer();
-  servers.push(contentServer, staticServer, resultServer);
-  await Promise.all([
-    contentServer.listeningPromise,
-    staticServer.listeningPromise,
-    resultServer.listeningPromise,
-  ]);
+async function initServers(onFinished, onError) {
+  let contentServer;
+  let staticServer;
+  let resultServer;
+  try {
+    contentServer = createContentServer(CONTENT_SERVER_PORT);
+    staticServer = launchStaticServer(currentDirectory, {
+      httpPort: PERF_TESTS_PORT,
+    });
+    resultServer = createResultServer(onFinished, onError);
```

Comment:
`createResultServer`  could also take port in argument, and the port can be defined as a constant similarly with `CONTENT_SERVER_PORT`  or `PERF_TESTS_PORT`

Reply:
I try something

Reply:
I've added both port and retry attempt to the page's fragment now